--- xf86-video-sis-0.9.1/configure.ac.0002~	2008-10-16 00:22:18.000000000 +0200
+++ xf86-video-sis-0.9.1/configure.ac	2013-03-26 16:49:51.559344387 +0100
@@ -58,7 +58,6 @@ XORG_DRIVER_CHECK_EXT(XINERAMA, xinerama
 XORG_DRIVER_CHECK_EXT(RANDR, randrproto)
 XORG_DRIVER_CHECK_EXT(RENDER, renderproto)
 XORG_DRIVER_CHECK_EXT(XV, videoproto)
-XORG_DRIVER_CHECK_EXT(XF86MISC, xf86miscproto)
 XORG_DRIVER_CHECK_EXT(DPMSExtension, xextproto)
 
 # Checks for pkg-config packages
--- xf86-video-sis-0.9.1/src/sis_driver.c.0002~	2013-03-26 16:49:51.558344399 +0100
+++ xf86-video-sis-0.9.1/src/sis_driver.c	2013-03-26 16:50:58.308509908 +0100
@@ -85,8 +85,6 @@
 #include "micmap.h"
 #include "mibank.h"
 #include "mipointer.h"
-#define _XF86MISC_SERVER_
-#include <X11/extensions/xf86misc.h>
 #include "edid.h"
 
 #define SIS_NEED_inSISREG
@@ -505,11 +503,6 @@ xf86DrvMsg(0, X_INFO, "
 	    pScrn->FreeScreen       = SISFreeScreen;
 	    pScrn->ValidMode        = SISValidMode;
 	    pScrn->PMEvent          = SISPMEvent; /*add PM function for ACPI hotkey,Ivans*/
-#ifdef X_XF86MiscPassMessage
-//   if(xf86GetVersion() >= XF86_VERSION_NUMERIC(4,3,99,2,0)) {
-//	       pScrn->HandleMessage = SISHandleMessage;
-//	    }
-#endif
 	    foundScreen = TRUE;
 	}
     #ifdef SISDUALHEAD
--- xf86-video-sis-0.9.1/src/sis_driver.h.0002~	2013-03-26 16:49:51.558344399 +0100
+++ xf86-video-sis-0.9.1/src/sis_driver.h	2013-03-26 16:49:51.560344374 +0100
@@ -791,10 +791,6 @@ static Bool SISHotkeySwitchMode(ScrnInfo
 #ifdef SISDUALHEAD
 static Bool	SISSaveScreenDH(ScreenPtr pScreen, int mode);
 #endif
-#ifdef X_XF86MiscPassMessage
-extern int	SISHandleMessage(int scrnIndex, const char *msgtype,
-				const char *msgval, char **retmsg);
-#endif
 static void     SISFreeScreen(int scrnIndex, int flags);
 static ModeStatus SISValidMode(int scrnIndex, DisplayModePtr mode,
 				Bool verbose, int flags);
--- xf86-video-sis-0.9.1/src/sis_utility.c.0002~	2008-09-02 16:16:32.000000000 +0200
+++ xf86-video-sis-0.9.1/src/sis_utility.c	2013-03-26 16:49:51.561344362 +0100
@@ -40,8 +40,6 @@
 #define NEED_EVENTS
 #include <X11/X.h>
 #include "dixstruct.h"
-#define _XF86MISC_SERVER_
-#include <X11/extensions/xf86misc.h>
 
 #include "sis_videostr.h"
 
@@ -262,9 +260,6 @@ typedef struct {
     int		 (*HandleSiSDirectCommand[SISCTRL_MAX_SCREENS])(xSiSCtrlCommandReply *);
 } xSiSCtrlScreenTable;
 
-#ifdef X_XF86MiscPassMessage
-int		SISHandleMessage(int scrnIndex, const char *msgtype, const char *msgval, char **retmsg);
-#endif
 void		SiSCtrlExtInit(ScrnInfoPtr pScrn);
 void		SiSCtrlExtUnregister(SISPtr pSiS, int index);
 
@@ -975,19 +970,6 @@ unsigned int SISVGADetected(ScrnInfoPtr
 }
 
 /***********************************
- *     MessageHandler interface    *
- *   (unused now; use extension)   *
- ***********************************/
-
-#ifdef X_XF86MiscPassMessage
-int
-SISHandleMessage(int scrnIndex, const char *msgtype, const char *msgval, char **retmsg)
-{
-    return BadMatch;
-}
-#endif
-
-/***********************************
  *   SiSCtrl extension interface   *
  ***********************************/
 
